Make the popup "switch to ALSA" dialogue a bit nicer

stuartlangridge opened the pull request:
Change the popup dialogue text to be framed as "if you have problems, switch to alsa" rather than "if you want problems, stay with pulseaudio". Also tidy up the dialogue box and format it properly (and fix a bug where it was missing a backslash in the `yad` command line), and frame responses to skew towards "do what I'm doing now", and word in terms of "Don't ask again" which is familiar from other dialogue types. Make the "always use PulseAudio" file be visible, so that if a user wants to undo their decision but can't remember how, they stand a chance of finding the file by poking around in `snap/APPNAME`.

I have not been able to test that this actually does the right thing, because doing so requires building and installing a whole snap, so that will need checking still of course.

stuartlangridge committed:
Rename settings file to something user-visible
If the user chooses to always use PulseAudio for this app, there should be a way to turn that off. Making the configuration file that sets this up have a user-visible understandable name means that the user can at least explore in snap/SNAPNAME and perhaps find it, even if they don't remember the message explaining how to do this.

stuartlangridge committed:
Rewrite popup message and tidy up
The popup message is rewritten to be framed as "maybe if you need to, switch to ALSA" rather than "you should switch to alsa but if you really must then you can stay using pulseaudio".
Also, tidy up the dialogue box a bit (give it better spacing and an icon, and some text formatting).

lucyllewy commented on snapcraft-assets/snap/command-chain/alsa-launch:
```diff
@@ -7,44 +7,53 @@ else
 
 popup() {
-  TEXT="$SNAP_READABLE_NAME requires access to the ALSA system.
+  TEXT="$SNAP_READABLE_NAME may have improved audio quality if granted permission to access the ALSA audio system.
```
I'm not sure audio quality is a problem that using alsa directly would necessarily resolve. It's more a case that you can't choose input and output devices when piping alsa through pulseaudio. (note that the application still uses alsa api but the alsa library is configured to forward that into pulseaudio)

```suggestion
  TEXT="$SNAP_READABLE_NAME may have better audio device support if granted permission to access the ALSA audio system.
```

stuartlangridge committed:
Update snapcraft-assets/snap/command-chain/alsa-launch

I'm completely cool with however you think this ought to be worded; I don't actually know what the problems _are_ that this wrapper is designed to resolve, so I'll happily accept this!

Co-authored-by: Daniel Llewellyn <daniel@bowlhat.net>

lucyllewy commented on snapcraft-assets/snap/command-chain/alsa-launch:
```diff
@@ -89,7 +98,10 @@ setup_alsa_override() {
 
 if [ -f "$SNAP_USER_DATA/.snapcraft-alsa" ]; then
-  . "$SNAP_USER_DATA/.snapcraft-alsa"
+  mv "$SNAP_USER_DATA/.snapcraft-alsa" "$SNAP_USER_DATA/Always use PulseAudio"
```
I'm wary of changing the filename here, and below, because people are already using this part and will be re-prompted even if they've previously told the script to always use pulseaudio...

lucyllewy replied:
oh, I see, you're renaming the old file.. ignore that then :-)

stuartlangridge replied:
indeed -- that's exactly why it does that :-)

lucyllewy commented:
Many Thanks, This LGTM :-)
